contrib/google.golang.org/grpc/server.go

```diff
@@ -84,6 +84,14 @@ func StreamServerInterceptor(opts ...Option) grpc.StreamServerInterceptor {
 				cfg.analyticsRate,
 			)
 			defer func() { finishWithError(span, err, cfg) }()
+			switch {
```

Comment:
I have to agree with @cgilmour here that it would be better to change `doClientRequest` and `startSpanFromContext` to take a `kind` argument and automatically attach the tag. It just seems like this belongs in there because that's where all tags are set.